### PR TITLE
Add oVirt ovirt_storage_domains and ovirt_storage_domains_facts modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- ovirt_storage_domains
- ovirt_storage_domains_facts
##### SUMMARY

This PR adds new modules to manage storage domains in oVirt.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @machacekondra for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
@mwperina @ryansb

File: cloud/ovirt/ovirt_storage_domains.py [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+        default: present
+    description:
+        description:
+            - "Description of the storage domain."
+    comment:
+        description:
+            - "Comment of the storage domain."
+    data_center:
+        description:
+            - "Datacenter name where storage domain should be attached."
```
Comment (mwperina) [inline cloud/ovirt/ovirt_storage_domains.py]: Datacenter -> Data center

File: cloud/ovirt/ovirt_storage_domains_facts.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        - "For example to search storage domain X from datacenter Y use following pattern:
+           name=X and datacenter=Y"
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about all storage domains named C<data**> from datacenter C(west):
```
Comment (mwperina) [inline cloud/ovirt/ovirt_storage_domains_facts.py]: I'd rephrase to: 'Gather facts about all storage domains which names start with C<data> and belong to data center C(west):'

**Review by mwperina — CHANGES_REQUESTED** [review]

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_storage_domains.py [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        elif state == 'unattached':
+            ret = storage_domains_module.create()
+            storage_domains_module.pre_remove(
+                storage_domain=storage_domains_service.service(ret['id']).get()
+            )
+            ret['changed'] = storage_domains_module.changed
+
+        module.exit_json(**ret)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_storage_domains.py]: Same here, please surface the full exception.

File: cloud/ovirt/ovirt_storage_domains_facts.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        module.exit_json(
+            changed=False,
+            ansible_facts=dict(
+                ovirt_storage_domains=[
+                    get_dict_of_struct(c) for c in storage_domains
+                ],
+            ),
+        )
+    except Exception as e:
+        module.fail_json(msg=str(e))
```
Comment (ryansb) [inline cloud/ovirt/ovirt_storage_domains_facts.py]: Same here, please surface the full exception.

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Review by mwperina — APPROVED** [review]
:+1:
